runtimes/huggingface/mlserver_huggingface/common.py

```diff
@@ -128,8 +137,207 @@ def load_pipeline_from_settings(hf_settings: HuggingFaceSettings) -> Pipeline:
     return pp
 
 
-class NumpyEncoder(json.JSONEncoder):
+def get_img_bytes(img: Image.Image, mime_type: Optional[str] = "png") -> bytes:
+    buf = io.BytesIO()
+    img.save(buf, format=mime_type)
+    return buf.getvalue()
+
+
+def b64encode_image(img: Image.Image, mime_type: Optional[str] = "png") -> str:
+    return base64.b64encode(get_img_bytes(img, mime_type=mime_type)).decode()
+
+
+def build_image(data: bytes) -> "Image.Image":
+    return Image.open(io.BytesIO(data))
+
+
+def serialize_conversation(obj: "Conversation") -> Dict:
+    return {
+        "conversation_id": str(obj.uuid),
+        "past_user_inputs": obj.past_user_inputs,
+        "generated_responses": obj.generated_responses,
+        "new_user_input": obj.new_user_input,
+    }
+
+
+def deserialize_conversation(data: Dict) -> "Conversation":
+    text = data.get("new_user_input", "")
+    c_id = data.get("conversation_id", "")
+    past = data.get("past_user_inputs", "")
+    conversation_id = uuid.UUID(c_id) if c_id else None
+    conversation = Conversation(
+        text=text,
+        conversation_id=conversation_id,
+    )
+    if past:
+        conversation.past_user_inputs = past
+    return conversation
+
+
+class CommonJSONEncoder(json.JSONEncoder):
     def default(self, obj):
         if isinstance(obj, np.ndarray):
             return obj.tolist()
-        return json.JSONEncoder.default(self, obj)
+        elif isinstance(obj, (np.float_, np.float16, np.float32, np.float64)):
+            return float(str(obj))
+        elif isinstance(obj, (np.int_, np.int8, np.int16, np.int32, np.int64)):
+            return int(obj)
+        elif isinstance(obj, Image.Image):
+            return b64encode_image(obj)
+        elif isinstance(obj, Conversation):
+            return serialize_conversation(obj)
+        else:
+            return json.JSONEncoder.default(self, obj)
+
+
+class MetaInfo:
```

Comment:
If this one is a static dict, we could also convert it into a constant dict (i.e. as opposed to a class).